### PR TITLE
[WFCORE-2409] kerberos-security-factory - GSS mechanisms by name

### PR DESCRIPTION
**Pull request by thaarok**
https://issues.jboss.org/browse/WFCORE-2409

Depends on https://github.com/wildfly-security-incubator/wildfly-core/pull/85
Depends and related to https://github.com/wildfly-security/wildfly-elytron/pull/735

### CODE REVIEW

File: elytron/src/test/resources/org/wildfly/extension/elytron/credential-security-factories.xml [hunk 1]
```diff
@@ -16,6 +16,7 @@
                                    server="false"
                                    obtain-kerberos-ticket="true"
                                    debug="true"
+                                   mechanisms="KRB5 KRB5LEGACY"
```
Comment (mchoma) [inline elytron/src/test/resources/org/wildfly/extension/elytron/credential-security-factories.xml]: Is it meant you can use either `mechanisms` or `mechanism-oids`? If yes, could be `alternatives` relation configured in model? Or does it have sense to configure both of them at same time?

Also consider renaming `mechanisms` -> `mechanism-names` to be consistent to `mechanism-oids`

Comment (thaarok) [reply elytron/src/test/resources/org/wildfly/extension/elytron/credential-security-factories.xml]: **It make sense to configure both**: for example following for KRB5+SPNEGO:
`mechanisms="KRB5" mechanism-oids="1.3.6.1.5.5.2"`

Comment (mchoma) [reply elytron/src/test/resources/org/wildfly/extension/elytron/credential-security-factories.xml]: And can I configure KRB5+SPNEGO also with:
* mechanisms="KRB5 SPNEGO"
or
* mechanism-oids="1.2.840.113554.1.2.2 1.3.6.1.5.5.2"
?

Comment (thaarok) [reply elytron/src/test/resources/org/wildfly/extension/elytron/credential-security-factories.xml]: Sure - just `mechanisms` are primary aimed to well known mechanisms and `mechanism-oids` allows to pass unusual/custom GSS mechanisms.

Comment (mchoma) [reply elytron/src/test/resources/org/wildfly/extension/elytron/credential-security-factories.xml]: +1 for option to set unusual/custom GSS mechanism. 

But could not that be achieved by possibility to pass pure OID into mechanisms? E.g. `mechanisms="KRB5 SPNEGO 1.2.3.4.5.6"`

If not, please:
*  add allowed values for `mechanisms` attribute
*  add into description, that `mechanism-oids` are like addition to `mechanisms` attribute - that first extends second. Something like

`elytron.kerberos-security-factory.mechanism-oids=Possibility to specify pure OIDs in addition to specified mechanism names in mechanisms attribute`

Comment (thaarok) [reply elytron/src/test/resources/org/wildfly/extension/elytron/credential-security-factories.xml]: @darranl Do we prefer one or two standalone attributes for this?
I am also not sure if allowedValues would be reasonable, as the list would duplicate content of property file in elytron and the same should be added into **x500-attribute-principal-decoder/attribute-name**.

Comment (mchoma) [reply elytron/src/test/resources/org/wildfly/extension/elytron/credential-security-factories.xml]: Seems to me subsystem allowed values are still somehow static enumeration - not changed dynamically based on intern implementation. See for example:
* server-ssl-context.protocol
* syslog-audit-log.format
* configurable-sasl-server-factory.filters

**Comment by thaarok** [comment]
Updated by comments - renamed `mechanisms` to `mechanism-names` and mentioned relation between `mechanism-names` and `mechanism-oids` in doc

**Comment by mchoma** [comment]
Nice. But still missing allowed values limitation.

**Comment by thaarok** [comment]
jaj... fixed - added allowed values

**Comment by mchoma** [comment]
thanxs, I have no other objections ;)

**Comment by darranl** [comment]
Going to need to hold this one back just until I tag Elytron tomorrow.

**Pull request merged**